### PR TITLE
Add setup-npmrc action

### PR DESCRIPTION
**Pull request by reakaleek**
## What does this PR do?

- Creates a GH Action named `setup-npmrc`, similarly to `withNpmrc` Jenkins Global Variable.

## Why is it important?

- Can be used in GH Actions workflows with npm registry operations like publishing, etc.

## Related issues
Closes #ISSUE

### CODE REVIEW

**Review by v1v — APPROVED** [review]
LGTM, thanks! I might need to change https://github.com/elastic/apm-agent-nodejs/blob/c641850eb73316e9ae0297e0365b727770fff25d/.github/workflows/release.yml#L71 as I used the old fashion way.

File: .github/actions/setup-npmrc/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+| Name            | Type   | Default                                          | Description                                     |
+|-----------------|--------|--------------------------------------------------|-------------------------------------------------|
+| `vaultRoleId`   | String |                                                  | The Vault role id.                              |
+| `vaultSecretId` | String |                                                  | The Vault secret id.                            |
+| `vaultUrl`      | String |                                                  | The Vault URL to connect to.                    |
+| `secret`        | String | `secret/apm-team/ci/elastic-observability-npmjs` | The Vault secret.                               |
+| `secretKey`     | String | `token`                                          | The Vault secret key.                           |
+| `registry`      | String | `registry.npmjs.org`                             | NPM Registry.                                   |
+| `npmrcfile`     | String | `.npmrc`                                         | Name of the file with the token.                |
```
Comment (v1v) [inline .github/actions/setup-npmrc/README.md:52]:
```suggestion
| `npmrcFile`     | String | `.npmrc`                                         | Name of the file with the token.                |
```

File: .github/actions/setup-npmrc/action.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    required: false
+  secretKey:
+    description: 'Vault secret key. (Optional. Default: token)'
+    default: token
+    required: false
+  registry:
+    description: 'NPM registry. (Optional. Default: registry.npmjs.org)'
+    default: registry.npmjs.org
+    required: false
+  npmrcfile:
```
Comment (v1v) [inline .github/actions/setup-npmrc/action.yml:27]: camelcase?

```suggestion
  npmrcFile:
```

File: .github/actions/setup-npmrc/action.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - uses: hashicorp/vault-action@v2.5.0
+      with:
+        url: ${{ inputs.vaultUrl }}
+        roleId: ${{ inputs.vaultRoleId }}
+        secretId: ${{ inputs.vaultSecretId }}
+        method: approle
+        secrets: |
+          ${{ inputs.secret }} ${{ inputs.secretKey }} | NPM_TOKEN
+    - run: |
+        echo "//${{ inputs.registry }}/:_authToken=${NPM_TOKEN}" > "${{ inputs.path }}/${{ inputs.npmrcfile }}"
```
Comment (v1v) [inline .github/actions/setup-npmrc/action.yml:47]:
```suggestion
        echo "//${{ inputs.registry }}/:_authToken=${NPM_TOKEN}" > "${{ inputs.path }}/${{ inputs.npmrcFile }}"
```

**Review by v1v — APPROVED** [review]

**Comment by reakaleek** [comment]
> Quick one, I don't know whether this is blocking anything, otherwise, if we agree to use the kebab-case style, shall that be reflected here so we don't need to support the deprecated fields?

I can change it to kebab-case. I'm currently working on the release pipeline in synthethics and I wanted to verify using the fork as reference first before merging and officially releasing it.

**Pull request merged**